modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -588,12 +622,18 @@ private MetadataContext getMetadataContext(MetadataCache cache)
       Optional<ConfigurationInstance> configuration = getConfiguration(fakeEvent);
 
       if (configuration.isPresent()) {
-        ConfigurationProvider configurationProvider = findConfigurationProvider()
+        ConfigurationProviderResolverWrapper configurationProviderResolver = findConfigurationProviderResolver()
             .orElseThrow(
                          () -> new MetadataResolvingException("Failed to create the required configuration for Metadata retrieval",
                                                               INVALID_CONFIGURATION));
 
-        if (configurationProvider instanceof DynamicConfigurationProvider) {
+        if (configurationProviderResolver.dependsOnEvent()) {
+          throw new MetadataResolvingException("Configuration used for Metadata fetch cannot depend on Event context",
+                                               INVALID_CONFIGURATION);
+        }
+
+        Optional<ConfigurationProvider> configurationProvider = resolveConfigurationProvider(configurationProviderResolver);
+        if (!configurationProvider.isPresent() || configurationProvider.get() instanceof DynamicConfigurationProvider) {
```

Comment:
Shouldn't this `instanceof` check be replaced by `ConfigurationProvider#isDynamic`?

Reply:
Changed to `ValueResolver#isDynamic`.